python/lsst/sims/catUtils/mixins/snObject.py

```diff
@@ -0,0 +1,718 @@
+        mwebv value calculated from the self.skycoord if not None, or set to
+        a value using self.set_MWebv. If neither of these are done, this value
+        will be None, leading to exceptions in extinction calculation.
+        Therefore, the value must be set explicitly to 0. to get unextincted
+        quantities.
+
+    Methods
+    -------
+
+    Examples
+    --------
+    >>> SNObject  = SNObject(ra=30., dec=60.)
+    >>> SNObject._ra
+    >>> 0.5235987755982988
+    >>> SNObject._dec
+    >>> 1.0471975511965976
+    """
+
+    def __init__(self, ra=None, dec=None, source='salt2-extended'):
+        """
+        Parameters
+        ----------
+        ra : float
+            ra of the SN in degrees
+        dec : float
+            dec of the SN in degrees
+        """
+
+        dust = sncosmo.OD94Dust()
+        sncosmo.Model.__init__(self, source=source, effects=[dust, dust],
+                               effect_names=['host', 'mw'],
+                               effect_frames=['rest', 'obs'])
+
+        # Current implementation of Model has a default value of mwebv = 0.
+        # ie. no extinction, but this is not part of the API, so should not
+        # depend on it, set explicitly in order to unextincted SED from
+        # SNCosmo. We will use catsim extinction from `lsst.sims.photUtils`.
+
+        self.ModelSource = source
+        self.set(mwebv=0.)
+
+        # ra and dec if passed are assumed to be in degrees and converted into
+        # radians.
+        self._ra = ra
+        self._dec = dec
+
+        # NB: More lines of code to support the possibility that ra, dec are
+        # not provided at instantiation, and default to None
+        self._hascoords = True
+        if self._dec is None:
+            self._hascoords = False
+        if self._ra is None:
+            self._hascoords = False
+
+        # Satisfied that coordinates provided are floats
+        if self._hascoords:
+            self.setCoords(ra, dec)
+
+        # For values of ra, dec, the E(B-V) is calculated directly
+        # from DustMaps
+        self.lsstmwebv = EBVbase()
+        self.ebvofMW = None
+        if self._hascoords:
+            self.mwEBVfromMaps()
+        return
+
+    @property
+    def SNstate(self):
+        """
+        Dictionary summarizing the property of the state. Can be used to
+        serialize to disk
+
+        Returns : Dictionary with values of parameters of the model.
+        """
+        statedict = dict()
+
+        # SNCosmo Parameters
+        statedict['ModelSource'] = self.source.name
+        for param_name in self.param_names:
+            statedict[param_name] = self.get(param_name)
+
+        # New Attributes
+        # statedict['lsstmwebv'] = self.lsstmwebv
+        statedict['_ra'] = self._ra
+        statedict['_dec'] = self._dec
+        statedict['MWE(B-V)'] = self.ebvofMW
+
+        return statedict
+
+    @staticmethod
+    def equivsncosmoParamDict(SNstate, SNCosmoModel):
```

Comment:
It can be done, but I don't think it is a good idea. I don't think the user will be using this function as part of the class a lot, but might use it as a static method without instantiating the class more often. Thus, allowing the user to set the dummy is more useful.